Gems/EMotionFX/Code/EMotionFX/Rendering/OpenGL2/Source/GLActor.cpp

```diff
@@ -121,34 +106,35 @@ namespace RenderGL
         mTexturePath        = texturePath;
 
         // get the number of nodes and geometry LOD levels
-        const uint32 numGeometryLODLevels   = actor->GetNumLODLevels();
-        const uint32 numNodes               = actor->GetNumNodes();
+        const size_t numGeometryLODLevels   = actor->GetNumLODLevels();
+        const size_t numNodes               = actor->GetNumNodes();
 
         // set the pre-allocation amount for the number of materials
-        mMaterials.Resize(numGeometryLODLevels);
+        mMaterials.resize(numGeometryLODLevels);
 
         // resize the vertex and index buffers
-        for (uint32 a = 0; a < 3; ++a)
+        for (AZStd::vector<VertexBuffer*>& vertexBuffers : mVertexBuffers)
         {
-            mVertexBuffers[a].Resize(numGeometryLODLevels);
-            mIndexBuffers[a].Resize(numGeometryLODLevels);
-            mPrimitives[a].Resize(numGeometryLODLevels);
-
-            // reset the vertex and index buffers
-            for (uint32 n = 0; n < numGeometryLODLevels; ++n)
-            {
-                mVertexBuffers[a][n] = nullptr;
-                mIndexBuffers [a][n] = nullptr;
-            }
+            vertexBuffers.resize(numGeometryLODLevels);
+            AZStd::fill(begin(vertexBuffers), end(vertexBuffers), nullptr);
```

Comment:
this can be done in 1 line:
`vertexBuffers.resize(numGeometryLODLevels, nullptr);`

same with `indexBuffers`